OpenRA.Mods.Common/Pathfinder/PathSearch.cs

```diff
@@ -29,45 +37,41 @@ static CellInfoLayerPool LayerPoolForWorld(World world)
 		{
-		}
+			var graph = new PathGraph(LayerPoolForWorld(world), locomotor, self, world, check, customCost, null, false);
+			var search = new PathSearch(graph, loc => 0, DefaultHeuristicWeightPercentage, targetPredicate);
+
+			foreach (var sl in froms)
+				if (world.Map.Contains(sl))
+					search.AddInitialCell(sl);
 
-		public static IPathSearch Search(World world, Locomotor locomotor, Actor self, BlockedByActor check, Func<CPos, bool> goalCondition)
-		{
-			var graph = new PathGraph(LayerPoolForWorld(world), locomotor, self, world, check);
-			var search = new PathSearch(graph);
-			search.isGoal = goalCondition;
-			search.heuristic = loc => 0;
 			return search;
 		}
 
-		public static IPathSearch FromPoint(World world, Locomotor locomotor, Actor self, CPos @from, CPos target, BlockedByActor check)
+		public static PathSearch ToTargetCell(
+			World world, Locomotor locomotor, Actor self, CPos from, CPos target, BlockedByActor check,
+			Func<CPos, int> customCost = null,
+			Actor ignoreActor = null,
+			bool inReverse = false)
 		{
-			return FromPoints(world, locomotor, self, new[] { from }, target, check);
+			return ToTargetCell(world, locomotor, self, new[] { from }, target, check, customCost, ignoreActor, inReverse);
 		}
 
-		public static IPathSearch FromPoints(World world, Locomotor locomotor, Actor self, IEnumerable<CPos> froms, CPos target, BlockedByActor check)
+		public static PathSearch ToTargetCell(
+			World world, Locomotor locomotor, Actor self, IEnumerable<CPos> froms, CPos target, BlockedByActor check,
+			Func<CPos, int> customCost = null,
+			Actor ignoreActor = null,
+			bool inReverse = false,
+			Func<CPos, int> heuristic = null,
+			int heuristicWeightPercentage = DefaultHeuristicWeightPercentage)
 		{
-			var graph = new PathGraph(LayerPoolForWorld(world), locomotor, self, world, check);
-			var search = new PathSearch(graph);
-			search.heuristic = search.DefaultEstimator(target);
-
-			// The search will aim for the shortest path by default, a weight of 100%.
-			// We can allow the search to find paths that aren't optimal by changing the weight.
-			// We provide a weight that limits the worst case length of the path,
-			// e.g. a weight of 110% will find a path no more than 10% longer than the shortest possible.
-			// The benefit of allowing the search to return suboptimal paths is faster computation time.
-			// The search can skip some areas of the search space, meaning it has less work to do.
-			// We allow paths up to 25% longer than the shortest, optimal path, to improve pathfinding time.
-			search.heuristicWeightPercentage = 125;
-
-			search.isGoal = loc =>
-			{
-				var locInfo = search.Graph[loc];
-				return locInfo.EstimatedTotalCost - locInfo.CostSoFar == 0;
-			};
+			var graph = new PathGraph(LayerPoolForWorld(world), locomotor, self, world, check, customCost, ignoreActor, inReverse);
+
+			heuristic = heuristic is null ? DefaultCostEstimator(locomotor, target) : heuristic;
```

Comment:
```suggestion
			heuristic = heuristic ?? DefaultCostEstimator(locomotor, target);
```

Reply:
For some reason when I couldn't use `??=` (lang version) I came up with a wordier expansion, this is obviously the better one.